src/main.rs

```diff
@@ -354,31 +358,39 @@ async fn entry(jobserver_client: LazyJobserverClient) -> Result<()> {
     })?;
 
     // Remove installed crates
-    let crate_names = CrateName::dedup(crate_names).filter_map(|crate_name| {
-        if opts.force {
-            Some((crate_name, None))
-        } else if let Some(records) = &metadata {
-            if let Some(metadata) = records.get(&crate_name.name) {
-                if let Some(version_req) = &crate_name.version_req {
-                    if version_req.is_latest_compatible(&metadata.current_version) {
-                        info!(
-                            "package {crate_name} is already installed and cannot be upgraded, use --force to override"
-                        );
-                        None
-                    } else {
-                        Some((crate_name, Some(metadata.current_version.clone())))
-                    }
-                } else {
-                    info!("package {crate_name} is already installed, use --force to override");
+    let crate_names = CrateName::dedup(crate_names)
+        .filter_map(|crate_name| {
+            match (
+                opts.force,
+                metadata.as_ref().and_then(|records| records.get(&crate_name.name)),
+                &crate_name.version_req,
+            ) {
+                (false, Some(metadata), Some(version_req))
+                    if version_req.is_latest_compatible(&metadata.current_version) =>
+                {
+                    debug!("Bailing out early because we can assume wanted is already installed from metafile");
+                    info!(
+                        "{} v{} is already installed, use --force to override",
+                        crate_name.name, metadata.current_version
+                    );
                     None
                 }
```

Comment:
```suggestion
                (false, Some(metadata), Some(version_req)) => {
                    if version_req.is_latest_compatible(&metadata.current_version) {
                        debug!("Bailing out early because we can assume wanted is already installed from metafile");
                        info!(
                            "{} v{} is already installed, use --force to override",
                            crate_name.name, metadata.current_version
                        );
                        None
                    } else {
                        Some((crate_name, Some(metadata.current_version.clone()))) 
                    }
                }
```